deps/rabbitmq_shovel_management/src/rabbit_shovel_mgmt_util.erl

```diff
@@ -0,0 +1,66 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2021 VMware, Inc. or its affiliates.  All rights reserved.
+%%
+
+-module(rabbit_shovel_mgmt_util).
+
+-export([status/2]).
+
+-import(rabbit_misc, [pget/2]).
+
+-include_lib("rabbitmq_management_agent/include/rabbit_mgmt_records.hrl").
+-include_lib("amqp_client/include/amqp_client.hrl").
+
+
+%% Allow users to see things in the vhosts they are authorised. But
+%% static shovels do not have a vhost, so only allow admins (not
+%% monitors) to see them.
+filter_vhost_user(List, _ReqData, #context{user = User = #user{tags = Tags}}) ->
+    VHosts = rabbit_mgmt_util:list_login_vhosts_names(User, undefined),
+    [I || I <- List, case pget(vhost, I) of
+                         undefined -> lists:member(administrator, Tags);
+                         VHost     -> lists:member(VHost, VHosts)
+                     end].
+
+status(ReqData, Context) ->
+    filter_vhost_user(
+        lists:append([status(Node) || Node <- [node() | nodes()]]),
+        ReqData, Context).
+
+status(Node) ->
+    case rpc:call(Node, rabbit_shovel_status, status, [], infinity) of
```

Comment:
Same thing as above, never use `infinity` timeouts, in particular when calling across the network.